gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-scim/src/main/java/io/gravitee/am/gateway/handler/scim/resources/bulk/BulkEndpoint.java

```diff
@@ -111,7 +113,9 @@ private BulkRequest checkBulkRequest(BulkRequest bulkRequest) {
         if (bulkRequest.getOperations().size() > BULK_MAX_REQUEST_OPERATIONS) {// TODO make this configurable in AM-3572
             throw TooManyOperationException.tooManyOperation(BULK_MAX_REQUEST_OPERATIONS);// TODO make this configurable in AM-3572
         }
-        if (bulkRequest.getOperations().stream().map(BulkOperation::getBulkId).distinct().count() != bulkRequest.getOperations().size()){
+        long uniqBulkId = bulkRequest.getOperations().stream().filter(op -> op.getMethod() == HttpMethod.POST || hasText(op.getBulkId())).map(BulkOperation::getBulkId).distinct().count();
+        long allBulkId = bulkRequest.getOperations().stream().filter(op -> op.getMethod() == HttpMethod.POST || hasText(op.getBulkId())).map(BulkOperation::getBulkId).count();
+        if (uniqBulkId != allBulkId){
```

Comment:
**just suggestion** for me would be more readable if first you collect all POST bulkIds, and then compare their count with distinct count